### ISSUE TITLE
[Accessibility Audit] Missing focus indicators for categories in the filter

### ISSUE THREAD

**Issue by SaptakS** [freedomofpress/pressfreedomtracker.us#2005]
The filter categories in the Incident database page don't have a visible focus or hover indicator.

<img width="660" height="324" alt="Image" src="https://github.com/user-attachments/assets/11d87a76-e833-4bf6-a528-35137a901e9a" />
